components/camel-hdfs/src/main/java/org/apache/camel/component/hdfs/HdfsConfiguration.java

```diff
@@ -171,33 +191,63 @@ private String getString(Map<String, Object> hdfsSettings, String param, String
 
     private List<HdfsProducer.SplitStrategy> getSplitStrategies(Map<String, Object> hdfsSettings) {
         List<HdfsProducer.SplitStrategy> strategies = new ArrayList<>();
-        for (Object obj : hdfsSettings.keySet()) {
-            String key = (String) obj;
-            if ("splitStrategy".equals(key)) {
-                String eit = (String) hdfsSettings.get(key);
-                if (eit != null) {
-                    String[] strstrategies = eit.split(",");
-                    for (String strstrategy : strstrategies) {
-                        String tokens[] = strstrategy.split(":");
-                        if (tokens.length != 2) {
-                            throw new IllegalArgumentException("Wrong Split Strategy " + key + "=" + eit);
-                        }
-                        HdfsProducer.SplitStrategyType sst = HdfsProducer.SplitStrategyType.valueOf(tokens[0]);
-                        long ssv = Long.valueOf(tokens[1]);
-                        strategies.add(new HdfsProducer.SplitStrategy(sst, ssv));
-                    }
+
+        splitStrategy = getString(hdfsSettings, "splitStrategy", kerberosNamedNodes);
+
+        if (Objects.nonNull(splitStrategy)) {
+            String[] strstrategies = splitStrategy.split(",");
+            for (String strstrategy : strstrategies) {
+                String[] tokens = strstrategy.split(":");
+                if (tokens.length != 2) {
+                    throw new IllegalArgumentException("Wrong Split Strategy [splitStrategy" + "=" + splitStrategy + "]");
                 }
+                HdfsProducer.SplitStrategyType sst = HdfsProducer.SplitStrategyType.valueOf(tokens[0]);
+                long ssv = Long.parseLong(tokens[1]);
+                strategies.add(new HdfsProducer.SplitStrategy(sst, ssv));
             }
         }
         return strategies;
     }
 
+    private List<String> getKerberosNamedNodeList(Map<String, Object> hdfsSettings) {
+        kerberosNamedNodes = getString(hdfsSettings, "kerberosNamedNodes", kerberosNamedNodes);
+        return Arrays.stream(kerberosNamedNodes.split(",")).distinct().collect(Collectors.toList());
+    }
+
+
+    Configuration getJAASConfiguration() {
```

Comment:
Is there a specific reason to move the jaasconfiguration here? It was a parameter at component level , now it's not even an endpoint option. It doesn't seem right.

Reply:
Good point, will redesign.

Reply:
Redesigned, please re-check.